lms/djangoapps/support/static/support/js/views/certificates.js

```diff
@@ -0,0 +1,151 @@
+
+                // Push a URL into history with the search query as a GET parameter.
+                // That way, if the user reloads the page or sends someone the link
+                // then the same search will be performed on page load.
+                window.history.pushState({}, window.document.title, url);
+
+                // Perform a search for the user's certificates.
+                this.disableButtons();
+                this.certificates.setUserQuery(query);
+                this.certificates.fetch({
+                    success: this.updateCertificates,
+                    error: this.handleSearchError
+                });
+            },
+
+            regenerateCertificate: function(event) {
+                var $button = $(event.target);
+
+                // Regenerate certificates for a particular user and course.
+                // If this is successful, reload the certificate results so they show
+                // the updated status.
+                this.disableButtons();
+                $.ajax({
+                    url: "/certificates/regenerate",
+                    type: "POST",
+                    data: {
+                        username: $button.data("username"),
+                        course_key: $button.data("course-key"),
+                    },
+                    context: this,
+                    success: function() {
+                        this.certificates.fetch({
+                            success: this.updateCertificates,
+                            error: this.handleSearchError,
+                        });
+                    },
+                    error: this.handleRegenerateError
+                });
+            },
+
+            updateCertificates: function() {
+                this.renderResults();
+                this.enableButtons();
+            },
+
+            handleSearchError: function(jqxhr) {
+                this.renderError(jqxhr.responseText);
+                this.enableButtons();
+            },
+
+            handleRegenerateError: function(jqxhr) {
+                // Since there are multiple "regenerate" buttons on the page,
+                // it's difficult to show the error message in the UI.
+                // Since this page is used only by internal staff, I think the
+                // quick-and-easy way is reasonable.
+                alert(jqxhr.responseText);
+                this.enableButtons();
+            },
+
+            triggerSearch: function() {
+                $('.certificates-form').submit();
+            },
+
+            getUserQuery: function() {
+                return $('.certificates-form input[name="query"]').val();
+            },
+
+            setUserQuery: function(query) {
+                $('.certificates-form input[name="query"]').val(query);
+            },
+
+            setResults: function(html) {
+                var $resultsDiv = $(".certificates-results", this.$el);
+                $resultsDiv.html(html);
```

Comment:
Nitpick: you don't really need to create the var because the selector isn't reused within the function.